### DISCUSSION TITLE
Data-Driven RANS Closure for NACA 0012 wake

### DISCUSSION THREAD

**Discussion by fabiofalletta04-ui** [lululxvi/deepxde#2089] [Q&A]
Hello everyone, I am a third-year Aerospace Engineering student at the Polytechnic University of Turin, completing my thesis on data-driven turbulence modeling for RANS correction (with a focus on anisotropy and turbulent kinetic energy correction for the wake of a NACA 0012 airfoil).
Since the literature on Turbulence Machine Learning is extremely heterogeneous (varying invariant bases, pure TBNNs vs KANs, different target metrics), I’m having a hard time finding a direct benchmark. I’d love to get the community's feedback on whether my current metrics are competitive with the current State of the Art.

To overcome the "information loss" of pure Tensor Basis Neural Networks (TBNNs) in strongly 2D wakes, I developed a cascaded (Dual-Brain) MLP architecture with specific Physics-Informed constraints:

Pre-trained on a universal dataset, fused with a 15% sparse data assimilation from the target NACA wake (with added Gaussian noise). Evaluated strictly on the unseen 85% of the wake domain (zero-shot).
Augmented standard invariants with specific non-invariant directional tensor components (adimensionalized via turbulent time scale) to act as a "spatial compass".
Instead of hard-clipping a posteriori, I embedded Lumley triangle realizability constraints directly into the loss function via soft asymmetric penalties (penalizing the optimizer if components drop below the physical limit of -1/3).
Replaced standard MSE with a zonally-weighted Huber Loss to handle RANS wall anomalies and avoid exploding gradients. Used LayerNorm + SiLU to prevent dead neurons.

The Results (Evaluated on the 85% unseen domain vs LES reference):

| Metric | RANS Baseline (k-w SST) | My Physics-Informed MLP |
| Global Anisotropy: | 100.3% | 10.9% |
| Global Reynolds Stresses: | 66.4% | 8.7% |
| Energy Magnitude Error (Frobenius): | N/A | 0.6% |

Specific component L2 errors:
uu: RANS 69.5% -> AI 8.4%
uv: RANS 94.2% -> AI 17.1%
vv: RANS 58.6% -> AI 7.4%
k: RANS 44.6% -> AI 5.9%

Considering the strict evaluation on the 85% unseen domain, is an 8.7% L2 error on dimensional Reynolds stresses considered SOTA for MLP-based models right now?
The global magnitude error is very low (0.6%), which I attribute to the strict physical constraints in the loss. Have you seen similar energy conservation metrics in recent papers?
Any feedback, thoughts on the architecture, or references to recent benchmark papers would be massively appreciated. Thanks!

<img width="2335" height="1450" alt="scia naca" src="https://github.com/user-attachments/assets/bc811568-5294-48dc-9aa0-6db60ce7c6d7" />
